### PR TITLE
Avoid undefined Bundler::Plugin::API::Source exception

### LINKED ISSUE #5409: Uninitialized constant Bundler::Plugin::API::Source.
Seems this issue has reappeared in bundler 1.14.x in combination with 2.4.x rubygems. See:

https://travis-ci.org/dekellum/syncwrap/jobs/197069321#L202

Its not an issue with either rubygems 2.6.x (2.6.6 in one env.), or with bundler 1.13.7 as I'm able to demonstrate here:

https://travis-ci.org/dekellum/syncwrap/jobs/197079264

I think this is very similar to reports and PR #4981.

If the rake project can reproduce the issue, we don't know via travis, given that its pegged to bundler 1.13.x:

https://github.com/ruby/rake/blob/34f5e2e27d2c49431b7c84d0787eaf6afcddf794/.travis.yml

Cc: @hsbt 

But here are some other reports of this with bundler 1.14.3:

* skywinder/github-changelog-generator#474
* tmuxinator/tmuxinator#503

### PR DESCRIPTION
**Pull request by segiddins**
Fixes #5409, with a test!

### CODE REVIEW

File: lib/bundler/plugin/api/source.rb [hunk 1]
```diff
@@ -287,6 +287,12 @@ def gem_install_dir
         def root
           Bundler.root
         end
+
+        # @private
+        # Returns true
```
Comment (indirect) [inline lib/bundler/plugin/api/source.rb]: lol this comment 😂

**Comment by indirect** [comment]
hooray, @bundlerbot r+

**Comment by abhitrivedi** [comment]
Thanks! Is there a date when 1.14.6 will be available on rubygems?

**Comment by segiddins** [comment]
Probably sometime this week

**Pull request merged**